tests/phpunit/class-test-core-sitemaps.php

```diff
@@ -334,7 +334,23 @@ public function test_robots_text() {
 		$robots_text = apply_filters( 'robots_txt', '', true );
 		$sitemap_string = 'Sitemap: http://' . WP_TESTS_DOMAIN . '/?sitemap=index';
 
-		$this->assertNotFalse( strpos( $robots_text, $sitemap_string ), 'Sitemap URL not included in robots text.' );
+		$this->assertContains( $sitemap_string, $robots_text, 'Sitemap URL not included in robots text.' );
+	}
+
+	/**
+	 * Test robots.txt output for a private site.
+	 */
+	public function test_robots_text_private_site() {
+		// Simulate private site (search engines discouraged).
+		update_option( 'blog_public', '0' );
+
+		$robots_text = apply_filters( 'robots_txt', '', true );
```

Comment:
shouldn't the last param be false?

Reply:
the site is not public in this case

Reply:
Right! Copy paste strikes again...

Although it shouldn't make a difference here, as our code should bail way earlier, i.e. `robots_txt` should never be filtered if the site is not public.

Reply:
Maybe a race condition? The filter is set by the test before and `update_option( 'blog_public', '0' );` does not unset the filter list?!?

Reply:
Yeah the plugin is loaded before all tests run, so this is to be expected

Reply:
But then `update_option( 'blog_public', '0' );` should be removed (because it is not really tested here) or the plugin has to be "reloaded" here, to test the option flag.

Reply:
Yep, I still need to get to that 👍

Reply:
hmmm... but if we want to test `update_option( 'blog_public', '0' );` we have to use `$robots_text = apply_filters( 'robots_txt', '', true );`... otherwise it does not test the option.

So we need:

```PHP
	/**
	 * Test robots.txt output for a private site.
	 */
	public function test_robots_text_private_site() {
                // @todo rest plugin

		// Simulate private site (search engines discouraged).
		update_option( 'blog_public', '0' );

		$robots_text = apply_filters( 'robots_txt', '', true );
		$sitemap_string = 'Sitemap: http://' . WP_TESTS_DOMAIN . '/?sitemap=index';

		// Simulate public site.
		update_option( 'blog_public', '1' );

		$this->assertNotContains( $sitemap_string, $robots_text );
	}
```

and

```PHP
```PHP
	/**
	 * Test robots.txt output for a private site.
	 */
	public function test_robots_text_private_site() {
		$robots_text = apply_filters( 'robots_txt', '', false );
		$sitemap_string = 'Sitemap: http://' . WP_TESTS_DOMAIN . '/?sitemap=index';

		$this->assertNotContains( $sitemap_string, $robots_text );
	}
```